sdk/core/src/policies/telemetry_policy.rs

```diff
@@ -0,0 +1,39 @@
+use crate::{Context, Policy, PolicyResult, Request, Response};
+
+use async_trait::async_trait;
+use std::sync::Arc;
+
+#[derive(Clone, Debug)]
+pub struct TelemetryOptions {
+    application_id: Option<String>,
+}
+
+impl TelemetryOptions {
+    pub fn new(application_id: Option<String>) -> Self {
```

Comment:
When would some want to create a `TelemetryPolicy` with an `application_id` set to `None`?

Reply:
This will actually be set by eventual `ClientOptions` traits, which each client must implement and take as input for instantiation (optional). This follows our SDK guidelines.

Customers (not client devs) can pass this in.